src/frontend/handlers/invoice.ts

```diff
@@ -37,59 +21,22 @@ export const getInvoiceHandler: RequestHandler<
       request.query.month
     ),
   ]);
-
-  let status;
-  let bannerClass;
-  if (lineItems.length === 0) {
-    status = "Invoice and events missing";
-    bannerClass = "warning";
-  } else if (
-    lineItems.find((lineItem) =>
-      WARNING_CODES.includes(lineItem.price_difference_percentage)
-    )
-  ) {
-    // We know at this point that at least one line item contains a warning, but
-    // we want to find the one with the highest priority warning.
-    const highestPriorityWarning: PercentageDiscrepancySpecialCase | undefined =
-      WARNINGS_BY_PRIORITY.find((warning) =>
-        lineItems.find(
-          (lineItem) =>
-            lineItem.price_difference_percentage === warning.magicNumber
-        )
-      );
-    if (!highestPriorityWarning) {
-      throw new Error("Couldn't find line item with warning");
-    }
-    status = highestPriorityWarning.bannerText;
-    bannerClass = "warning";
-  } else if (
-    lineItems.find((lineItem) => +lineItem.price_difference_percentage >= 1)
-  ) {
-    status = "Invoice above threshold";
-    bannerClass = "error";
-  } else if (
-    lineItems.find((lineItem) => +lineItem.price_difference_percentage <= -1)
-  ) {
-    status = "Invoice below threshold";
-    bannerClass = "notice";
-  } else {
-    status = "Invoice within threshold";
-    bannerClass = "payable";
-  }
-
+  console.log("line items", lineItems);
+  const invoiceBanner = getInvoiceBanner(lineItems);
+  console.log("invoiceBanner", invoiceBanner);
   const reconciliationRows = getReconciliationRows(lineItems);
 
   response.render("invoice.njk", {
-    classes: bannerClass,
     invoice: {
-      status,
       vendorName: config.vendorName,
       contractName: config.contractName,
       contractId: request.query.contract_id,
       year: request.query.year,
       prettyMonth: MONTHS[Number(request.query.month) - 1],
       month: request.query.month,
     },
+    bannerClass: invoiceBanner.bannerClass,
+    invoiceStatus: invoiceBanner.status,
     reconciliationRows,
```

Comment:
Maybe the `invoice: {...}` wrapper can go away?  I put it in to try to keep things tidy, but the page is for showing an invoice so all these variables are going to pertain to the invoice.  Or everything should go inside it.  I don't feel strongly either way.